src/asm.rs

```diff
@@ -29,16 +29,41 @@ pub mod x64 {
     fn gen_program(w: &mut impl Write, program: codegen::Program) -> Result<()> {
         w.write_str("\t.intel_syntax noprefix\n\n")?;
 
-        for f in program.functions.into_iter() {
-            gen_function(w, f)?;
+        for entry in program.top_level.into_iter() {
+            match entry {
+                codegen::TopLevel::Fun(f) => gen_function(w, f)?,
+                codegen::TopLevel::Static(v) => gen_static(w, v)?,
+            }
         }
 
         w.write_str("\n\t.section .note.GNU-stack,\"\",@progbits\n")?;
         Ok(())
     }
 
+    fn gen_static(w: &mut impl Write, var: codegen::StaticVariable) -> Result<()> {
+        if var.global {
+            w.write_fmt(format_args!("\t.globl {}\n", var.identifier))?;
+        }
+        let init_value = var.init.unwrap_or(0);
+        if init_value == 0 {
+            w.write_fmt(format_args!("\t.bss\n"))?;
+        } else {
+            w.write_fmt(format_args!("\t.data\n"))?;
+        }
+        // TODO: Unhardcode 4 here - take symbol info into this pass and calculate alignment
+        w.write_fmt(format_args!("\t.align 4\n"))?;
+        w.write_fmt(format_args!("{}:\n", var.identifier))?;
+        w.write_fmt(format_args!("\t.long {}\n", init_value))?;
```

Comment:
Zero initialized data should be declared with `.zero` instead of `.long`
```suggestion
        if init_value == 0 {
            w.write_fmt(format_args!("\t.zero 4\n"))?;
        } else {
            w.write_fmt(format_args!("\t.long {}\n", init_value))?;
        }
```

Reply:
U right